openassessment/xblock/ui_mixins/mfe/mixin.py

```diff
@@ -62,61 +63,49 @@ def get_block_info(self, data, suffix=""):  # pylint: disable=unused-argument
-            - If submitted, load fir
-        - If user provides jumpable step
-            - Check if we can get to that step
-            - If we can, go to that step
-            - Otherwise, raise an error
+        Get data for the user / step of the ORA, based on the following modes:
+
+        1) If no step provided, refresh progress but don't return any response.
+        2) If step provided, validate that we can get to that step and return the appropriate response
         """
+        # Query workflow step here only once to avoid duplicate workflow updates
         workflow_step = self.workflow_data.status or "submission"
-        jump_step = suffix
-        serializer_context = {"step": workflow_step}
-
-        # Allow jumping to a specific step, within our allowed steps
-        if jump_step:
-            jumpable_steps = ("submission", "peer", "grades")
-            if jump_step not in jumpable_steps:
-                raise JsonHandlerError(404, f"Invalid jump to step: {jump_step}")
-            if self._can_jump_to_step(workflow_step, jump_step):
-                serializer_context.update({"jump_to_step": suffix})
-            else:
-                raise JsonHandlerError(400, f"Cannot jump to step: {jump_step}")
+        active_step = suffix
 
-        # Determine which mode we are viewing in, since data comes from different sources
-        # View submission, submitted or draft
-        if workflow_step == "submission" or jump_step == "submission":
-            serializer_context.update({"view": "submission"})
-        # View the current assessment step
-        else:
-            serializer_context.update({"view": "assessment"})
+        # Validate that any active step is a valid step
+        if suffix and suffix not in handler_suffixes.STEP_SUFFIXES:
+            raise OraApiException(400, error_codes.INCORRECT_PARAMETERS, f"Invalid step name: {active_step}")
 
-        page_context = PageDataSerializer(self, context=serializer_context)
-        return page_context.data
+        serializer_context = {"active_step": None, "workflow_step": workflow_step}
 
-    def _can_jump_to_step(self, workflow_step, jump_step):
-        """ A helper to determine if we can jump to a step or not """
+        # For the general case, just return refreshed page data, without a response
+        if not active_step:
+            return PageDataSerializer(self, context=serializer_context).data
 
-        step_name_mappings = {
-            "submission": "submission",
-            "peer": "peer",
-            "grades": "done"
-        }
+        # Check to see if user can access this step
+        if not self._user_has_reached_step(workflow_step, active_step):
+            raise OraApiException(400, error_codes.INACCESSIBLE_STEP, f"Inaccessible step: {active_step}")
+
+        # If they have access to this step, return the associated data
+        serializer_context.update({"active_step": active_step})
```

Comment:
nit: this can just be serializer_context["active_step"] = active_step, we're just doing the one field